applications/asset_tracker_v2/src/modules/cloud_module.c

```diff
@@ -1048,53 +1252,17 @@ static void on_all_states(struct cloud_msg_data *msg)
 		}
 	}
 
+#if defined(CONFIG_NRF_CLOUD_AGNSS)
+	if (IS_EVENT(msg, location, LOCATION_MODULE_EVT_AGNSS_NEEDED)) {
```

Comment:
There is some handling of `LOCATION_MODULE_EVT_AGNSS_NEEDED` event both in `on_all_states()` and `on_sub_state_cloud_disconnected()`. This is probably wrong and might be a bit more complicated to handle this perfectly as there is no obvious good place for either code piece. The code can obviously be copied into multiple states but hopefully we are able to find a proper place for it.

Reply:
Moved both `LOCATION_MODULE_EVT_AGNSS_NEEDED` and `LOCATION_MODULE_EVT_PGPS_NEEDED` handling from `on_all_states()` to `on_sub_state_cloud_connected()`.

When not connected to the cloud, A-GNSS request is buffered in `on_sub_state_cloud_disconnected()` and `on_state_lte_disconnected()`. As before, the buffered request is handled when moving from cloud disconnected to cloud connected state.

Reply:
Yep, this should be fine.